dspace/modules/api/src/main/java/org/datadryad/anywhere/AssociationAnywhere.java

```diff
@@ -58,31 +60,38 @@ public static void main(String[] argv)
         Options options = new Options();
 
         options.addOption("i", "customer id", true, "customer id");
-        options.addOption("u", "update credit", false, "update credit");
-        options.addOption("d", "deduct credit", false, "deduct credit");
+        options.addOption("p", "data package id", false, "package id");
+        options.addOption("u", "update customer settings", false, "update customer settings");
+        options.addOption("t", "tally credit", false, "tally credit");
         options.addOption("l", "list customer", false, "list customer");
         CommandLine line = parser.parse(options, argv);
 
-        if(line.hasOption("u"))
-        {
-            Context context = new Context();
-            if(line.hasOption("i"))
-                updateConcept(context, line.getOptionValue("i"));
-            else
-                updateConcept(context);
-
-        }
-        else if(line.hasOption("d")){
-            deductCredit(line.getOptionValue("i"));
-        }
-        else if(line.hasOption("l")){
-            System.out.print(printDocument(loadCustomerInfo(line.getOptionValue("i"))));
-        }
-        else if(line.hasOption("i")){
-            //load credit
-            String credit = getCredit(line.getOptionValue("i"));
-            System.out.println("credit : "+ credit);
-        }
+	Context context = new Context(); 
+	try {
+	    if(line.hasOption("u")) {
+		log.debug("aa updating credits");
+		if(line.hasOption("i"))
+		    updateConcept(context, line.getOptionValue("i"));
+		else
+		    updateConcept(context);
+	    }
+	    else if(line.hasOption("t")){
+		log.debug("aa tallying credits");
+		tallyCredit(context, line.getOptionValue("i"), line.getOptionValue("p"));
+	    }
+	    else if(line.hasOption("l")){
+		log.debug("aa list customer");
+		System.out.print(printDocument(loadCustomerInfo(context, line.getOptionValue("i"))));
+	    }
+	    else if(line.hasOption("i")) {
+		log.debug("aa listing credits");
+		//load credit
+		String credit = getCredit(context, line.getOptionValue("i"));
+		System.out.println("credit : "+ credit);
+	    }
+	} finally {
```

Comment:
This is in a main() method -- so any exception will come out to the command line. How would a catch help?